src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -97,7 +97,9 @@ public async Task InitializeAsync()
 
             // Check to see if a solution is already open. If so we set _solutionOpened to true so that subsequent projects added to 
             // this solution are processed.
-            if (ErrorHandler.Succeeded(solution.GetProperty((int)__VSPROPID4.VSPROPID_IsSolutionFullyLoaded, out object isFullyLoaded)) && isFullyLoaded is bool && (bool)isFullyLoaded)
+            if (ErrorHandler.Succeeded(solution.GetProperty((int)__VSPROPID4.VSPROPID_IsSolutionFullyLoaded, out object objIsFullyLoaded)) &&
+                objIsFullyLoaded is bool isFullyLoaded &&
+                isFullyLoaded)
```

Comment:
Pretty sure this can be simplified further to just `objIsFullyLoaded is true` instead of the last two conditions.

[Seems to work without issue](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAIAYAEqCMBuAWAChjUBmbHANmwCZKB2TAb2M3ewtxtQBZMAsgEMAllAAUAShZsOcgG5CwmIZgC8mcQHsARgCsApgGNgkgERCAznABmZwkTkKlmHes27DJycEgGHThyygSI2mqoilpi+EAaSwU6sjoGYAJC4AJziFpiRUADkwNF+ZpJ4cglyAL6VQckhYeJukcWx8fWJtYGZ2ToQRc2WpQEpNR0VHWNVQA)

Reply:
You're right. That is cleaner.

I was curious what the `is` syntax actually does. Turns out it calls `object.Equals(object,object)` which means [it boxes the `true`](https://sharplab.io/#v2:C4LglgNgPgAgzAAhgRgGxIEwIMIIN4CwAUAqUoiugEYD2NECAsgBQ1UBWApgMbAI0BKfMTKikAdn4IwAZwTAATgFdOAbhFkAvsU1A===) (IL_0001), which means a small heap allocation.

Is this code called a lot? Where do we sit on the performance vs. readability spectrum in general?

Reply:
This is called once throughout lifetime of VS, readability wins out. That boxing sucks though for that pattern, but won't show up here at all.

Reply:
Fixed, thanks.

Reply:
TIL, thanks. I thought the constant pattern did a type check first. I am less enamored by it.